packages/bot/src/stores/toolbar-store.js

```diff
@@ -0,0 +1,283 @@
+import filesaver                        from 'file-saver';
+import {
+    observable,
+    action,
+    computed,
+}                                       from 'mobx';
+import { createErrorAndEmit }           from '../utils/error';
+import { observer as globalObserver }   from '../utils/observer';
+import {
+    cleanUpOnLoad,
+    fixCollapsedBlocks,
+    backwardCompatibility,
+    fixArgumentAttribute,
+    addDomAsBlock,
+}                                       from '../scratch/utils';
+import { translate }                    from '../utils/tools';
+import googleDrive                      from '../utils/integrations/googleDrive';
+
+export default class ToolbarStore {
+    constructor(flyout) {
+        this.flyout = flyout;
+    }
+
+    @observable is_toolbox_open = false;
+    @observable is_saveload_modal_open = false;
+    @observable is_save_modal = true;
+    @observable saveload_type = 'local';
+    @observable file_name = 'Untitled Bot';
+    @observable is_google_drive_connected = false;
+
+    @action.bound onRunClick = () => {
+        // TODO
+    }
+
+    @action.bound onStartClick = () => {
+        this.is_toolbox_open = !this.is_toolbox_open;
+        const toolbox = Blockly.derivWorkspace.toolbox_;
+
+        if (this.is_toolbox_open) {
+            toolbox.open();
+        } else {
+            toolbox.close();
+        }
+    }
+
+    @action.bound onSearchBlur = () => {
+        this.on_search_focus = false;
+    }
+
+    @action.bound onSearch = ({ search }) => {
+        const toolbox = Blockly.derivWorkspace.toolbox_;
```

Comment:
- You can destructure this, e.g. `const { toolbox_: toolbox } = Blockly.derivWorkspace;`
- Since we're accessing it a couple times we could set a reference to the toolbox in this store's constructor.